render.c

```diff
@@ -36,9 +36,31 @@ static void move_to(cairo_t *cairo, double x, double y, int scale) {
 	cairo_move_to(cairo, x * scale, y * scale);
 }
 
-static void set_rectangle(cairo_t *cairo, double x, double y, double width, double height,
-		int scale) {
-	cairo_rectangle(cairo, x * scale, y * scale, width * scale, height * scale);
+static void set_rounded_rectangle(cairo_t *cairo, double x, double y, double width, double height,
+		int scale, double radius) {
+	if (width == 0 || height == 0) {
+		return;
+	}
+	x *= scale;
+	y *= scale;
+	width *= scale;
+	height *= scale;
+	double degrees = M_PI / 180.0;
+
+	if (radius > width / 2) {
+		radius = 0;
```

Comment:
Should these snap to `width / 2` (or height) rather than throwing it out entirely?

Reply:
I prefer to widen the rectangle to accomodate the radius (I had done this in the past, it must have been undone at some point). This should only come up in very thin progress bars, and I think it's better to have the progress bar be slightly wider than expected than to render it with the wrong radius (which would leave a sliver of the background color in between the border and the progress bar).